src/bundle-manifest.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * A bundle strategy function is used to transform an array of bundles.
+ */
+export type BundleStrategy = (bundles: Bundle[]) => Bundle[];
+
+/**
+ * A mapping of entrypoints to their full set of transitive dependencies,
+ * such that a dependency graph `a->c, c->d, d->e, b->d, b->f` would be
+ * represented `{a:[a,c,d,e], b:[b,d,e,f]}`.  Please note that there is an
+ * explicit identity dependency (`a` depends on `a`, `b` depends on `b`).
+ */
+export type TransitiveDependenciesMap = Map<UrlString, Set<UrlString>>;
+
+/**
+ * Defining this URL type to make it clear which strings represent URLs.
+ */
+export type UrlString = string;
+
+/**
+ * A bundle is a grouping of files which serve the need of one or more
+ * entrypoint files.
+ */
+export class Bundle {
+  public entrypoints: Set<UrlString>;
+  public files: Set<UrlString>;
+  constructor(entrypoints?: Set<UrlString>, files?: Set<UrlString>) {
+    this.entrypoints = entrypoints || new Set<UrlString>();
+    this.files = files || new Set<UrlString>();
+  }
+}
+
+/**
+ * Given an index of files and their dependencies, produce an array of bundles,
+ * where a bundle is defined for each set of dependencies.
```

Comment:
"where a bundle is defined for each set of dependencies" -> "where a bundle is defined for each set entrypoints that has unique dependencies"? I'm hunting for a clear description of this...